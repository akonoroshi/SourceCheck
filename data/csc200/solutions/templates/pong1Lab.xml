<project name="Pong In Lab" app="iSnap 2.1, http://go.ncsu.edu/isnap" version="1" assignment="pong1Lab"><data>{}</data><notes></notes><thumbnail>data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAKAAAAB4CAYAAAB1ovlvAAAGCklEQVR4Xu2d3SttaRzHf5wZJHOYjJeIg0beI0ODXCgyF9PEUF5SuHCh5EKUC1yYm2mSmqLGxYTmZkqJixkX/gATivEy3i6OtxxJwo1TOJmeZ7KbfdDZe2n2b57n911X9t5rref7/X0+1l6nnL397u7u7ggbJsA0AT8IyDR5LKsnAAEhAusEICDr+LE4BIQDrBOAgKzjx+IQEA6wTgACso4fi0NAOMA6AQjIOn4sDgHhAOsEICDr+LE4BIQDrBOAgKzjx+IQEA6wTgACso4fi0NAOMA6AQjIOn57F/fz83Mr99TfPUNAex1gbfYsAVdXV2lvb49ubm5YSzy1eGBgICUnJ1NKSsr/Mh9CETkWcGxsjIqLi30yw9dTg/Tx2jR5+t9SQl9+Qq/6/6CLiwudb2FhgRoaGnySFYt4NwFHAp6fn9Px8fGDlc7Ozig8PFw/v7KyQtnZ2frnw8NDiouLezTZ9PQ0VVZWul67vr6mgIAAt32vfyyjzz/9yO05/1dfkP/LaLpd+/3R84b8cEAbGxuu19LT072bDPb2yQQcCTg4OEhlZWWugHV1ddTW1kYxMTHU399P6q25vr6erq6uqK+vjzo6OmhoaIjUfomJiTQwMKD3U4+rq6spNTWVIiIiqKWlhdbX12lubo5KS0upqqpKrxH689f0WbC/20BCvt+ld2/W6e3QN08KqH4J7rf7XwafTBWLeDwBRwL29PS4XbWam5tpfHyc1NUrJCSEcnJyqLa2ljo7OykzM5MmJyf1fZjab2RkhMLCwigpKYkKCwtpbW2N1NVJSVpRUaGDt7e30+7uLg0PD+vHEb98+4iAr+ndwZ/09qd/JH1/U1fAxcVF19P5+fkeDwU7+m4CjgScmpqiqKgoV8qJiQmqqanRV7Gmpiba3Nykrq4uKi8vp4yMDGptbdVSKQHz8vL01a+3t5dGR0epsbGRoqOjtYwlJSXU3d2t38YLCgooNjZWrxHza80DAV9kfEUvEr+k69++e1JAdSW934qKinw3Vazk8QQcCajOPjs7S+pfme9vCQkJ+vnt7e0Hrymx1H2it1vKdAOFBLi/BX/oHHe9f9HS0pLeTd2z/vs+80PH4nXfTcCxgArqzMyM25Xwv4qtpA4ODvb49Le3t7Szs6P3Pzo60rcDQUFBHh+PHX03AccC+i4iVrJ5AhDQZroGdIOABkCyOSIEtJmuAd0goAGQbI4IAW2ma0A3CGgAJJsjQkCb6RrQDQIaAMnmiBDQZroWdcOf5FsE08QqENBEahZlhoAWwTSxCgQ0kZpFmSGgRTBNrAIBTaRmUWYIaBFME6tAQBOpWZQZAloE08QqENBEahZlhoAWwTSxCgQ0kZpFmSGgRTBNrAIBTaRmUWYIaBFME6tAQBOpWZQZAloE08QqENBEahZlhoAWwTSxCgQ0kZpFmSHgM2DOz8/rD1hXH7qJzdkEIKCzuVFkZCSdnp7qo+Pj42l/f9/hmWQfBgEd8F9eXqbc3Fy3Iw8ODp78xgAHS4g5BAI6QK0+8T8rK8vtyJOTE31VxObdBCCgd/Ny7Z2WlkZbW1v6sfrGAPWtANi8nwAE9H5mriMuLy/1z6Ghoc84i+xDIaBs/uztISA7AtkBIKBs/uztISA7AtkBIKBs/uztISA7AtkBIKBs/uztISA7AtkBIKBs/uztISA7AtkBIKBs/uztISA7AtkBIKBs/uztISA7AtkBIKBs/uztISA7AtkBIKBs/uztISA7AtkBIKBs/uztISA7AtkBIKBs/uztISA7AtkBIKBs/uztISA7AtkBIKBs/uztISA7AtkBIKBs/uztISA7AtkBIKBs/uztISA7AtkBIKBs/uztISA7AtkBIKBs/uztISA7AtkBIKBs/uztISA7AtkBIKBs/uztISA7AtkBIKBs/uztISA7AtkBIKBs/uztISA7AtkBIKBs/uztISA7AtkBIKBs/uztISA7AtkBIKBs/uztISA7AtkBIKBs/uztISA7AtkBIKBs/uztISA7AtkBIKBs/uztISA7AtkBIKBs/uztISA7AtkBIKBs/uzt/waQhwnGIscPHgAAAABJRU5ErkJggg==</thumbnail><stage name="Stage" width="480" height="360" costume="0" tempo="60" threadsafe="false" lines="round" ternary="true" codify="false" inheritance="false" sublistIDs="false" scheduled="false" id="1"><pentrails>data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAeAAAAFoCAYAAACPNyggAAAOhUlEQVR4Xu3VwQkAAAjEMN1/abewn7jAQRC64wgQIECAAIF3gX1fNEiAAAECBAiMAHsCAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQIHLFxAWmhEwHPAAAAAElFTkSuQmCC</pentrails><costumes><list id="2"></list></costumes><sounds><list id="3"></list></sounds><variables></variables><blocks></blocks><scripts></scripts><sprites><sprite name="Right Paddle" idx="1" x="215" y="100" heading="90" scale="0.3" rotation="1" draggable="true" costume="2" color="80,80,80" pen="tip" id="8"><costumes><list id="9"><item><costume name="Untitled" center-x="18" center-y="83" image="data:," id="10"/></item><item><costume name="Paddle" center-x="20" center-y="96" image="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACgAAADACAYAAAB26SGCAAABWElEQVR4Xu3dQQqCYBRG0efO2lkuzaWF0UBoeIN+4TT/4HW8jt1m5jkzj1nzd2yfA/c175v9dgeekv/WvN5wP8HVUiRYnwhBglWg7jVIsArUvQYJVoG61yDBKlD3GiRYBepegwSrQN1rkGAVqHsNEqwCda9BglWg7jVIsArUvQYJVoG61yDBKlD3GiRYBepegwSrQN1rkGAVqHsNEqwCda9BglWg7jVIsArUvQYJVoG61yDBKlD3GiRYBepegwSrQN1rkGAVqHsNEqwCda9BglWg7jVIsArUvQYJVoG61yDBKlD3GiRYBepegwSrQN1rkGAVqHsNEqwCda9BglWg7jVIsArUvQYJVoG61yDBKlD3GiRYBepegwSrQN1rkGAVqHsNEqwCda9BglWg7jVIsArUvQYJVoG61yDBKlD3GiRYBer+q8Fjge/IXm/wkvzkEa/6aeD3n3sBcNIi0aXV8YwAAAAASUVORK5CYII=" id="11"/></item></list></costumes><sounds><list id="12"></list></sounds><blocks></blocks><variables></variables><scripts><script x="143.00000000000003" y="78"><block id="90" s="receiveGo"></block><block id="91" s="setScale"><l>30</l></block><block id="92" s="gotoXY"><l>215</l><l>0</l></block><block id="1246" s="doForever"><script><block id="94" s="doIf"><block id="1274" s="reportAnd"><block id="1562" s="reportLessThan"><block id="1583" s="yPosition"></block><block id="1564" s="reportMouseY"></block></block><block id="95" s="reportLessThan"><block id="96" s="yPosition"></block><l>150</l></block></block><script><block id="97" s="changeYPosition"><l>5</l></block></script></block><block id="99" s="doIf"><block id="1275" s="reportAnd"><block id="1563" s="reportLessThan"><block id="1565" s="reportMouseY"></block><block id="1584" s="yPosition"></block></block><block id="100" s="reportGreaterThan"><block id="101" s="yPosition"></block><l>-150</l></block></block><script><block id="102" s="changeYPosition"><l>-5</l></block></script></block></script></block></script></scripts></sprite><sprite name="Ball" idx="2" x="0" y="0" heading="67" scale="0.4" rotation="1" draggable="true" costume="1" color="161.262,237.15,0" pen="tip" id="61"><costumes><list id="62"><item><costume name="Untitled" center-x="18" center-y="18" image="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACQAAAAkCAYAAADhAJiYAAACUUlEQVRYR83YS6gOcRjH8c/ZWSlFrgt2lBTZOBQpSexcc8utWCBLJ5cFTuyIDTkot1x3pChRLhtRUuxYEKKUlR09mjlN0/vOzHvOO+95n93M/J/f851n/v//PP+nx9BsCZZiEsZjQiLzDd/xBQ/xqFX5nhYcerEXyzC6ot9vPMBpvKjiUwVoOvqwuYpgwZjLOI4PRTplQNtwYZggefftuNhMswjoGA60GSaV68fBRtrNgAYQb5LaK8wdJlxeIzK/I6/ZCCiyEtnphEWWIluDlgdaj2udIMnE2IDr6XUWKJbyJ4zpMNAvTEVsEbJAsST3JzDvMLNmsGyME8nWMggUhB9rBiiTnxZfKM3QIRwp86j5+WEcTYFiW59Xc8Ay+ZfoDaDJ+Fw2ukPPpwTQatzqUMCyMGsCaDfOlI3s0PM9AVTnP6vV9+gPoPON/imtKrVp/EAAncS+NgkOV+ZUAMXuHLt0N1hfAG3BpW6gwdYAWognXQK0KN2pf2DsCEP9xLgUKArwTSMMdCUOEinQWtxIgP7mypI6ObOx1uFmth56hvl1Ri/Qfo4F+QJtOe6NENAK3M8DxfUdrOww1F2sSmPmi/xRiPTNSQbEzG/36stqvk6myZ9mQHF/RgJVd7EfxX3M2ffZL9LsoLgYVzGxps/3FRvxOK9fdJSOojugouvRTotyOWAaHirKmg0BchY720R0DruKtKoAhf/s5NwU5e5Q7HZSUbwpc64KlOrMShpW0bSKn3KRPU2aVdGwelsGUrTKqvrG0TsmfaOWXkza/0fjVu0ft3taxhKU2YsAAAAASUVORK5CYII=" id="63"/></item></list></costumes><sounds><list id="64"></list></sounds><blocks></blocks><variables></variables><scripts><script x="362" y="87"><block id="66" s="receiveGo"></block><block id="67" s="doSetVar"><l>right points</l><l>0</l></block><block id="68" s="setScale"><l>40</l></block><custom-block id="764" s="reset ball">undefined</custom-block></script><script x="624" y="88"><block id="1019" s="receiveKey"><l><option>space</option></l></block><block id="72" s="doForever"><script><block id="73" s="forward"><l>5</l></block><block id="74" s="doIf"><block id="75" s="reportTouchingObject"><l>Right Paddle</l></block><script><block id="76" s="setHeading"><block id="77" s="reportDifference"><l>360</l><block id="78" s="direction"></block></block></block><block id="719" s="doChangeVar"><l>right points</l><l>1</l></block></script></block><block id="81" s="doIf"><block id="82" s="reportGreaterThan"><block id="83" s="xPosition"></block><l>230</l></block><script><block id="1312" s="doSetVar"><l>right points</l><l>0</l></block><custom-block id="788" s="reset ball">undefined</custom-block></script></block><block id="358" s="bounceOffEdge"></block></script></block></script></scripts></sprite><watcher scope="Ball" s="yPosition" style="normal" x="10" y="31.000001999999995" color="74,108,212" hidden="true"/><watcher scope="Ball" s="direction" style="normal" x="10" y="78.600006" color="74,108,212" hidden="true"/><watcher var="right points" style="normal" x="10" y="57.600004" color="243,118,29"/></sprites></stage><hidden></hidden><headers></headers><code></code><blocks><block-definition s="reset ball" type="command" category="motion" guid="93a415d3-1683-4232-a96c-572b79f5d017" isImported="false"><header></header><code></code><translations></translations><inputs></inputs><script><block id="69" s="gotoXY"><l>0</l><l>0</l></block><block id="70" s="setHeading"><block id="71" s="reportRandom"><l>30</l><l>150</l></block></block><block id="1021" s="doStopThis"><l><option>this script</option></l></block></script></block-definition></blocks><variables><variable name="right points"><l>0</l></variable></variables><editing></editing></project>