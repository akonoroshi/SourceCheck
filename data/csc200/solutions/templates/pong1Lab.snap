snapshot {
    @defHint PADDLE_POSITION {
        :`The Paddle position should be set to (215, 0). To set its position use "go to x: __ y: __" block (under Motion). Note that this position should be set at beginning of the Paddle code`
    }
    @defHint FOREVER_LOOP {
        :`To make the Paddle keep moving during the game, you need its code to run forever. The "forever" block (under Control) runs the code you put inside of it forever.`
    }
    @defHint IF_CONDITION {
        :`The Paddle should move up if the the mouse is above the Paddle's "y position" (under Motion) and down if it is lower.`
    }
    @defHint AND_OPERATOR {
        :`You can combine two conditions for one "if" block by using the "and" block (under Operators).`
    }
    @defHint KEY_PRESSED {
        :`Use the "key pressed" block (under Sensing) to check whether a key is pressed. You can choose which key from the dropdown list.`
    }
    @defHint COMPARE {
        :`Use "<" or ">" blocks (under Operators) to check if the Paddle's "y position" block is above or below the "mouse y" position (under Sensing).`
    }
    @defHint Y_VALUE1 {
        :`The Paddle's "y position" (under Motion) must be less than 150. If it exceeds 150 the Paddle will be off screen.`
    }
    @defHint Y_VALUE2 {
        :`The Paddle's "y position" (under Motion) must be greater than -150. If it is less than -150 the Paddle will be off screen.`
    }
    @defHint PADDLE_UP {
        :`Use the Paddle's "change y by __" block (under Motion) with a positive value to move the Paddle up, or with a negative value to move it down.`
    }
    @defHint BALL_FOREVER {
        :`To make the Ball keep moving during the game, you need its code to run forever. The "forever" block (under Control) runs the code you put inside of it forever.`
    }
    @defHint MOVE_BALL {
        :`Use "move" block (under Motion) with a positive value to allow the ball to move forward.`
    }
    @defHint TOUCH_PADDLE {
        :`Use "touching" block (under Sensing) to detect if the Ball sprite has touched the Paddle. Note that this block should be used in an "if" condition.`
    }
    @defHint TOUCH_BACK_EDGE {
        :`If the ball touches the right edge (x > 230), the player's score should go to 0 and the ball should reset.`
    }
    @defHint BACK_EDGE_X {
        :`You can use the Ball's "x position" (under Motion) and the ">" block (under Operators) to check if it is touching the right edge (x > 230).`
    }
    @defHint BALL_DIRECTION {
        :`Use the "point in direction __" block (under Motion) point the Ball in to the opposite direction when it hits the Paddle. To make it bounce, the ball should face 360 - its current "direction" (under Motion).`
    }
    @defHint SCORE_CHANGE {
        :`The player's score should start at 0. It should go up by one if the player "scores" (hitting the left edge), and reset to 0 if the ball hits the right wall.`
    }
    @defHint BOUNCE_OFF {
        :`To make the ball bounces off any edge, use "if on edge, bounce" (under Motion)`
    }
    @defHint BALL_RESET {
        :`You have to "reset" the Ball, after setting its points to zero. You can find the "reset" custom block under Motion.`
    }
    stage {
        sprite {
            script {
                receiveGo
                setScale(literal)
                @hint PADDLE_POSITION
                gotoXY(literal, literal)
                @hint FOREVER_LOOP
                doForever(script {
                    @or {
                        @inline {
                            @hint IF_CONDITION
                            doIf(@or {
                                    @hint COMPARE
                                    reportLessThan(@hint IF_CONDITION yPosition, @hint COMPARE reportMouseY)
                                    reportAnd(@hint COMPARE reportLessThan(@hint IF_CONDITION yPosition, @hint COMPARE reportMouseY), @hint Y_VALUE1 reportLessThan(yPosition, literal:150))
                                }, script {
                                @hint PADDLE_UP
                                changeYPosition(literal)
                            })
                            @hint IF_CONDITION
                            doIf(@or {
                                    @hint COMPARE
                                    reportLessThan(@hint COMPARE reportMouseY, @hint IF_CONDITION yPosition)
                                    reportAnd(@hint COMPARE reportLessThan(@hint COMPARE reportMouseY, @hint IF_CONDITION yPosition), @hint Y_VALUE2 reportLessThan(literal:-150, yPosition))
                                }, script {
                                @hint PADDLE_UP
                                changeYPosition(literal)
                            })
                        }
                        setYPosition(reportMouseY)
                    }
                })
            }
        }
        sprite {
            script {
                receiveGo
                @anyOrder {
                    @hint SCORE_CHANGE
                    doSetVar(varMenu:rightPoints, literal)
                    setScale(literal)
                    evaluateCustomBlock:resetBall
                }
            }
            script {
                receiveKey(literal)
                @optional { setHeading(@anything) }
                @hint BALL_FOREVER
                doForever(script {
                    @anyOrder {
                        @hint MOVE_BALL
                        forward(literal)
                        @defBlock BOUNCE {
                            @hint TOUCH_PADDLE
                            reportTouchingObject(literal), script {
                                @anyOrder {
                                    @hint BALL_DIRECTION
                                    setHeading(@hint BALL_DIRECTION reportDifference(literal, @hint BALL_DIRECTION direction))
                                    @hint SCORE_CHANGE
                                    doChangeVar(varMenu:rightPoints, literal)
                                }
                            }
                        }
                        @defBlock RESET {
                            @hint TOUCH_BACK_EDGE
                            doIf(@or {
                                @hint BACK_EDGE_X 
                                reportLessThan(literal, @hint BACK_EDGE_X xPosition)
                                reportAnd(@hint BACK_EDGE_X  reportLessThan(literal, @hint BACK_EDGE_X xPosition), reportTouchingObject(literal))
                            }, script {
                                @hint SCORE_CHANGE
                                doSetVar(varMenu:rightPoints, literal)
                                @hint BALL_RESET
                                evaluateCustomBlock:resetBall
                            })
                        }
                        @or {
                            @inline {
                                @hint TOUCH_PADDLE
                                doIf(@block BOUNCE)
                                @block RESET
                            }
                            @hint TOUCH_PADDLE
                            doIfElse(@block BOUNCE, script {
                                @block RESET
                            })
                        }
                    }
                    @hint BOUNCE_OFF
                    bounceOffEdge
                })
            }
        }
    }
    customBlock:resetBall {
        script {
            gotoXY(literal, literal)
            setHeading(reportRandom(literal, literal))
            doStopThis(literal)
        }
    }
    varDec:rightPoints
}
